src/ui/server/controllers/insurance/export-contract/how-much-is-the-agent-charging/index.test.ts

```diff
@@ -0,0 +1,276 @@
+    });
+  });
+
+  describe('FIELD_ID', () => {
+    it('should have the correct ID', () => {
+      const expected = FIXED_SUM_AMOUNT;
+
+      expect(FIELD_ID).toEqual(expected);
+    });
+  });
+
+  describe('get', () => {
+    it('should call api.keystone.APIM.getCurrencies', async () => {
+      await get(req, res);
+
+      expect(getCurrenciesSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it('should render template', async () => {
+      res.locals.application = mockApplication;
+
+      await get(req, res);
+
+      const generatedPageVariables = pageVariables(referenceNumber, allCurrencies, String(fixedSumCurrencyCode));
+
+      const { DYNAMIC_PAGE_TITLE } = generatedPageVariables;
+
+      const expectedVariables = {
+        ...singleInputPageVariables({
+          FIELD_ID,
+          PAGE_CONTENT_STRINGS: {
+            ...PAGE_CONTENT_STRINGS,
+            PAGE_TITLE: DYNAMIC_PAGE_TITLE,
+          },
+          BACK_LINK: req.headers.referer,
+        }),
+        ...generatedPageVariables,
+        userName: getUserNameFromSession(req.session.user),
+        application: mapApplicationToFormFields(mockApplication),
+      };
+
+      expect(res.render).toHaveBeenCalledWith(TEMPLATE, expectedVariables);
+    });
+
+    describe('when there is no application', () => {
+      beforeEach(() => {
+        delete res.locals.application;
+      });
+
+      it(`should redirect to ${PROBLEM_WITH_SERVICE}`, async () => {
+        await get(req, res);
+
+        expect(res.redirect).toHaveBeenCalledWith(PROBLEM_WITH_SERVICE);
+      });
+    });
+
+    describe('api error handling', () => {
+      describe('when the get currencies API call fails', () => {
+        beforeEach(() => {
+          getCurrenciesSpy = mockSpyPromiseRejection;
+          api.keystone.APIM.getCurrencies = getCurrenciesSpy;
+        });
+
+        it(`should redirect to ${PROBLEM_WITH_SERVICE}`, async () => {
+          await get(req, res);
+
+          expect(res.redirect).toHaveBeenCalledWith(PROBLEM_WITH_SERVICE);
+        });
+      });
+
+      describe('when the get currencies response does not return a populated array', () => {
+        beforeEach(() => {
+          getCurrenciesSpy = jest.fn(() => Promise.resolve(mockCurrenciesEmptyResponse));
+          api.keystone.APIM.getCurrencies = getCurrenciesSpy;
+        });
+
+        it(`should redirect to ${PROBLEM_WITH_SERVICE}`, async () => {
+          await get(req, res);
+
+          expect(res.redirect).toHaveBeenCalledWith(PROBLEM_WITH_SERVICE);
+        });
+      });
+    });
+  });
+
+  describe('post', () => {
+    beforeEach(() => {
+      getCurrenciesSpy = jest.fn(() => Promise.resolve(mockCurrenciesResponse));
+      api.keystone.APIM.getCurrencies = getCurrenciesSpy;
+    });
+
+    const validBody = {
+      [FIELD_ID]: mockApplication.exportContract.agent.service.charge[FIELD_ID],
+    };
+
+    describe('when there are no validation errors', () => {
```

Comment:
do we need a test to show that getCurrencies has not been called?

Reply:
Great spot, added.